scripts/lib/CIME/XML/env_batch.py

```diff
@@ -299,16 +299,23 @@ def submit_jobs(self, case, no_batch=False, job=None, batch_args=None):
                 if prereq is None or job == firstjob:
                     prereq = True
                 else:
+                    if dry_run:
```

Comment:
Oh, I see what you mean. We want to preserve the other prereqs besides the BUILD_COMPLETE one so that we get an accurate list of which jobs will be run.